Feature#7 frontend index

AngelBFdev opened the pull request:
# Description
- Header design 
- Form design and layout for the Index page
- Clock design 
- Clock functionality with javascript

## Screenshots
Mockup
![image](https://user-images.githubusercontent.com/50933273/148308059-0027411e-e897-422f-873f-e2bb51abadce.png)
Frontend
![image](https://user-images.githubusercontent.com/50933273/148308581-4abfda4f-14af-42b2-8ec2-55bc5c3c2d7b.png)

DerekAyala committed:
Custom properties added on a sass file

- the colors of the mockups
- font family

Co-authored-by: Anisbeflo <abejarano0@ucol.mx>
Co-authored-by: MauricioNC <mauricio_navarro0@ucol.mx>

DerekAyala committed:
normalize added

Co-authored-by: Anisbeflo <abejarano0@ucol.mx>
Co-authored-by: MauricioNC <mauricio_navarro0@ucol.mx>

DerekAyala committed:
Global properties added Sass

- new box sizing
- font size 62.5% to use rems and not px
- font family added

Co-authored-by: Anisbeflo <abejarano0@ucol.mx>
Co-authored-by: MauricioNC <mauricio_navarro0@ucol.mx>

Commit:
Home page frontend - in process

- Header design completed
- Clock: design for the clock area
- Form: design and layout for check in and check out registration

Co-authored-by: Anisbeflo <abejarano0@ucol.mx>
Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>

Commit:
Home page frontend completed

- Header design completed
- Form desing and layout completed
- Clock deisgn and functionality completed

JavaScript for clock functionality
- scritp to show the clock in real time using a function to get the time
  and setInterval.

Co-authored-by: Anisbeflo <abejarano0@ucol.mx>
Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>

kevinnio commented:
I think it should be `Check In` and `Check Out`.

kevinnio commented on Gemfile:
```diff
@@ -38,7 +40,7 @@ group :development do
   # Display performance information such as SQL time and flame graphs for each request in your browser.
   # Can be configured to work on production as well see: https://github.com/MiniProfiler/rack-mini-profiler/blob/master/README.md
-  gem 'rack-mini-profiler', '~> 2.0'
+  # gem 'rack-mini-profiler', '~> 2.0's
```
Did you comment this line out because it was failing? Looks like that `s` at the end might have been the culprit. 🤔

MauricioNC replied:
mmm, I don't know why this line is commented, no one us commented this line but when we ran the app we had no problemas. We will search more about this gem.

kevinnio replied:
This gem is not required for the app to work, but it can be helpful while developing it. I'll let you guys decide if you want to keep it or not, so if you decide to remove it, please remove the line completely instead of just commenting it out.

kevinnio commented on Gemfile:
```diff
@@ -3,6 +3,8 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '3.0.2'
 
+gem 'wdm', '>= 0.1.0'
```
Is this https://rubygems.org/gems/wdm? Why did you need to add a low-level gem like that to the project? 🤔

MauricioNC replied:
When we tryied to make the base project we got some issues at the momment to run the rake db:migrate command, searching on internet we foun that installing this gem the problem it would be solved and yes, now we deleted this gem and ran the app and everything is okay... so, I don't know what was the problem initially🤔

kevinnio replied:
I see. Well, if the app works just as well without it, then let's remove it. The fewer the gems, the better! Helps to keep [technical debt](https://www.productplan.com/glossary/technical-debt/) down.

kevinnio commented on Gemfile.lock:
```diff
@@ -201,6 +207,7 @@ GEM
     zeitwerk (2.5.1)
 
 PLATFORMS
+  x64-mingw32
```
...ohh, maybe `wdm` was added because someone is using `mingw` to develop and something was broken? 🤔

MauricioNC replied:
We are developing this project on windows, and yeah, I am using mingw, maybe that was the problem with the database creation, but now that we delete this gem, we have no errors, I don't know if it's necessary have installed this gem for future possible errors

kevinnio commented on app/assets/stylesheets/src/base/global.css.scss:
```diff
@@ -0,0 +1,24 @@
+  scroll-padding-top: 0rem;
+}
+
+*,
```
Is this style required? I think `normalize.css` already includes a style similar to this.

MauricioNC replied:
This is resolved with normalize.css from bootstrap

kevinnio commented on app/assets/stylesheets/src/base/normalize.css.scss:
```diff
@@ -0,0 +1,349 @@
+/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
```
Including this stylesheet is ok, but usually, we add the `.min` version of third-party stylesheets like this one, since they're smaller and faster to download for users.

MauricioNC replied:
This is resolved with normalize.css from bootstrap

kevinnio commented on app/javascript/packs/clock.js:
```diff
@@ -0,0 +1,31 @@
+    sec = sec < 10 ? "0" + sec : sec;
+
+    let currentTime = hour + ":" + min;
+    let meriedem = am_pm;
```
I get `currentTime` but looks like `meriedem` and `seconds` are not really useful variables here. 🤔

AngelBFdev replied:
You're right, we are working on it. Thanks

kevinnio commented on app/javascript/stylesheets/application.scss:
```diff
@@ -1 +1 @@
-@import 'bootstrap'
+@import 'bootstrap';
```
Oh, just noticed you're importing `bootstrap` here. As far as I'm aware, Bootstrap includes its own version of `normalize.css` with it, so maybe adding the stylesheet directly is not really needed? 🤔

MauricioNC replied:
Oooh yeah, you're right, we don't need our normalize.css

kevinnio commented on package.json:
```diff
@@ -10,11 +10,11 @@
     "turbolinks": "^5.2.0",
-    "webpack": "^4.46.0",
-    "webpack-cli": "^3.3.12"
+    "webpack": "4.46.0",
```
Any particular reason to change the version constraints of these packages? 🤔

MauricioNC replied:
We got an error when ran the app, the error shows something like webpacker compilation is not configured as true, and that we need to install weback, I do not remember the error specifically, but we found as a solution run the command: rails webpacker:install

kevinnio reviewed (commented):
Left some questions and suggestions.

YilKanda reviewed (commented):
My comment is directed regarding your commits:

1. I notice that you have a commit that says -in process- and one that says -complete-. It is not recommended to do this, since it remembers that your commits must be significant, that is, each commit contributed to a change within your project.

![imagen](https://user-images.githubusercontent.com/12243331/148312061-43ea95f2-481d-48e5-9ac9-bb443adba46a.png)

2. Try to be more specific in your commits, that is, what change was made? Why was it done? How does this change modify the direction of the project? Do not assume the reviewer understands what the original problem was, ensure you add it. Do not assume your code is self-explanatory.

![imagen](https://user-images.githubusercontent.com/12243331/148312315-361a7bd9-ba60-448e-83e7-aea8b413b829.png)

Here's a great template of a good commit message originally written by Tim pope:

```
Capitalized, short (50 chars or less) summary

More detailed explanatory text, if necessary.  Wrap it to about 72
characters or so.  In some contexts, the first line is treated as the
subject of an email and the rest of the text as the body.  The blank
line separating the summary from the body is critical (unless you omit
the body entirely); tools like rebase can get confused if you run the
two together.

Write your commit message in the imperative: "Fix bug" and not "Fixed bug"
or "Fixes bug."  This convention matches up with commit messages generated
by commands like git merge and git revert.

Further paragraphs come after blank lines.

- Bullet points are okay, too

- Typically a hyphen or asterisk is used for the bullet, followed by a
  single space, with blank lines in between, but conventions vary here

- Use a hanging indent

If you use an issue tracker, add a reference(s) to them at the bottom,
like so:

Resolves: #123
```

3. Specify the type of commit:


    **feat:** The new feature you're adding to a particular application
    **fix:** A bug fix
    **style:** Feature and updates related to styling
    **refactor:** Refactoring a specific section of the codebase
    **test:** Everything related to testing
    **docs:** Everything related to documentation
    **chore:** Regular code maintenance.[ You can also use emojis to represent commit types]

Commit:
Config modified

- normalize.css: this file was deleted because we already have this file
  using bootstrap, and it's not necessary include our normalize styles.

- clock.js: It was delete the meriedem and seconds variables because we
  already have a am_pm and sec variables that do the same.

- Gemfile: It was deleted the 's' in the rack-mini-profile line, it was
  not necessary to install this gem because comes commented by default,
  and the wdm gem was installed to solve some issues at the moment to
  run rake db:migrate command.

Co-authored-by: Anisbeflo <abejarano0@ucol.mx>
Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>

JuanequeX commented on app/assets/stylesheets/src/base/variables.css.scss:
```diff
@@ -0,0 +1,10 @@
+$secondary_color: #00E8B0;
+$table_cell: #C4C4C4;
+//font family
+$font: Consolas,monaco,monospace; 
```
Did can add a blank line here

MauricioNC replied:
Done.

JuanequeX commented on app/assets/stylesheets/src/base/variables.css.scss:
```diff
@@ -0,0 +1,10 @@
+
```
Delete this extra space

MauricioNC replied:
Done.

JuanequeX commented on app/assets/stylesheets/src/layout/form.css.scss:
```diff
@@ -0,0 +1,78 @@
+@import "../base/variables.css";
+
+.main {
+
```
Delete this extra space

MauricioNC replied:
Done.

JuanequeX commented on app/assets/stylesheets/src/layout/header.css.scss:
```diff
@@ -0,0 +1,29 @@
+
+
+.navbar {
+
```
Delete this extra space

MauricioNC replied:
Done.

JuanequeX commented on app/javascript/packs/clock.js:
```diff
@@ -0,0 +1,29 @@
+    document.getElementById("meriedem").innerHTML = am_pm;
+    document.getElementById("seconds").innerHTML = sec;
+}
+showTime();
```
Add a blank line here

MauricioNC replied:
Done.

Commit:
Structure SASS modified

- It was corrected the structure SASS for the header.css.scss,
  form.css.scss and vairables.css.scss.

Corrections:
 - Add an extra blank line into the file variable.css.scss
 - Delete an extra space into the file variables.css.scss
 - Delete an extra space into the file header.css.scss
 - Add an extra blank line into the file clock.js

Co-authored-by: Anisbeflo <abejarano0@ucol.mx>
Co-authored-by: DerekAyala <derek.ayala12345@hotmail.com>

YilKanda reviewed (approved):
Check The indentation of your files.

JuanequeX reviewed (approved):
LGTM
